### PR TITLE
feat(locking): ui for new unlock mechanic TRNG918

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Extend StepInspector UI to show the property unlock list.

### CODE REVIEW

File: Editor/UI/Drawers/LockableObjectsDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Innoactive.Creator.Core;
+using Innoactive.Creator.Core.Properties;
+using Innoactive.Creator.Core.RestrictiveEnvironment;
+using Innoactive.Creator.Core.SceneObjects;
+using NAudio.Wave;
+using UnityEditor;
+using UnityEngine;
+using Object = UnityEngine.Object;
```
Comment (Gusinuhe) [inline Editor/UI/Drawers/LockableObjectsDrawer.cs:12]:
```suggestion
﻿using System;
using System.Linq;
using Innoactive.Creator.Core;
using Innoactive.Creator.Core.Properties;
using Innoactive.Creator.Core.SceneObjects;
using UnityEditor;
using UnityEngine;
```

File: Runtime/Properties/PropertyReflectionHelper.cs [hunk 2]
```diff
@@ -17,9 +17,21 @@ namespace Innoactive.Creator.Core
     /// </summary>
     internal static class PropertyReflectionHelper
     {
-        /// <summary>
-        /// Extracts all <see cref="LockablePropertyData"/> from given Condition.
-        /// </summary>
+        public static List<LockablePropertyData> ExtractLockablesFromStep(IStepData data)
```
Comment (Gusinuhe) [inline Runtime/Properties/PropertyReflectionHelper.cs:20]: So, no more documentation? ☹️

File: Runtime/Step.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@
 using Innoactive.Creator.Core.EntityOwners;
 using Innoactive.Creator.Core.EntityOwners.FoldedEntityCollection;
 using Innoactive.Creator.Core.RestrictiveEnvironment;
+using Innoactive.Creator.Core.SceneObjects;
```
Comment (Gusinuhe) [inline Runtime/Step.cs:13]: This has nothing to do here 👮‍♂️

File: Runtime/Tabs/DynamicTab.cs [hunk 4]
```diff
@@ -20,6 +20,10 @@ public void SetValue(object value)
             setter(value);
         }
 
+        public void OnSelected() { }
+
+        public void OnUnselect() { }
```
Comment (Gusinuhe) [inline Runtime/Tabs/DynamicTab.cs:25]: Missing docs

File: Runtime/Properties/PropertyReflectionHelper.cs (hunk 2 above)
Comment (SimonTheSourcerer) [reply Runtime/Properties/PropertyReflectionHelper.cs:20]: I guess the name of the method is more then sufficient to explain what is happening :P

File: Runtime/Tabs/TabsGroup.cs [hunk 5]
```diff
@@ -9,7 +9,20 @@ internal class TabsGroup : ITabsGroup
...
+                return selected;
+            }
+            
+            set
+            {
+                selected = value;
+            }
+        }
+
+        private int selected;
```
Comment (tomwim) [inline Runtime/Tabs/TabsGroup.cs:25]: Two options:
1. Don't think we need this anymore. Was introduced in a previous iteration where logic was within the getter and setter of the property but that's not the case anymore
2. Trigger OnSelected und OnUnselected in here which makes it less errorprone when actually using it -> I do not have to change the value and trigger the method

File: Runtime/Step.cs [hunk 6]
```diff
@@ -72,7 +73,8 @@ public EntityData()
             {
                 Tabs = new TabsGroup(
                     new DynamicTab(new GUIContent("Behaviors"), () => Behaviors, value => Behaviors = (IBehaviorCollection)value),
-                    new DynamicTab(new GUIContent("Transitions"), () => Transitions, value => Transitions = (ITransitionCollection)value)
+                    new DynamicTab(new GUIContent("Transitions"), () => Transitions, value => Transitions = (ITransitionCollection)value),
```
Comment (tomwim) [inline Runtime/Step.cs:75]: Still not the biggest fan of UI within logic but not part of this ticket

File: Runtime/Tabs/TabsGroup.cs (hunk 5 above)
Comment (tomwim) [reply Runtime/Tabs/TabsGroup.cs:25]: Compare with TabsGroupDrawer change value (line 52)

Comment (SimonTheSourcerer) [reply Runtime/Tabs/TabsGroup.cs:25]: removed it

**Pull request merged**